### PR TITLE
rework stories and layout

### PR DESCRIPTION
**Pull request by sroussey**
I should have not mixed these together, but...

This PR:

1. Moves stories to their own folder
2. Reworks layout a bit. You need to register a layout before using it. This has a few upsides: 
  - The lib does not include dagre and all the requirements, if they are not used.
  - Adding new layouts required changing this lib
  - Easier to force different layout in the layout example rather than just the one, all without needing to change the lib itself
3. Moves the example node editor out of NodeEditor and into the stories examples.

### CODE REVIEW

**Comment by sroussey** [comment]
This does have a bug though...

In storybook, the first item "Docs" now shows nothing. Not sure how that works.

File: lib/layout/layout.ts [hunk 1]
```diff
@@ -25,11 +23,19 @@ function computeLayout(
...
 }
 
+const layoutEngines: Record<LayoutEngine, LayoutFunc> = {}
+
+export function registerLayoutEngine(
+  engine: LayoutEngine,
+  layoutFn: LayoutFunc,
+) {
+  layoutEngines[engine] = layoutFn
+}
```
Comment (clarkmcc) [inline lib/layout/layout.ts:33]: I'm pretty sure that this function needs to be exported from the `lib/index.ts` file in order to be consumed by projects when used as an external dependency.

**Review by clarkmcc — CHANGES_REQUESTED** [review]
Love it! One change requested in order for this to be used as an external dependency.

File: lib/layout/layout.ts (hunk 1 above)
Comment (sroussey) [reply lib/layout/layout.ts:33]: Indeed!

**Review by clarkmcc — APPROVED** [review]

**Pull request merged**